errors/invalidargerror.go

```diff
@@ -0,0 +1,13 @@
+package errors
+
+type InvalidArgError struct {
+	s string
+}
+
+func (e InvalidArgError) Error() string {
+	return e.s
+}
+
+func NewInvalidArgError(message string) error {
+	return InvalidArgError{s: message}
+}
```

Comment:
A couple thoughts on errors:

1. Unlike Java, you don't have to define them in their own files. We often put the errors relevant to a given bit of code in the same file (often at the bottom) as that code.
1. If your error is just a wrapper for a single type, you can declare it as an "alias" for that type to simplify usage a bit:

```go
type InvalidArgError string

func (e InvalidArgError) Error() string {
	return string(e)
}

// To create this error elsewhere in the code:
return InvalidArgError("My error message")
```

Reply:
Ah ok